### PR TITLE
add CI, fix #71 and minor architecture improvements

### LINKED ISSUE #71: tensorflow-gpu is in extras_require, but is contained in the 2.x tensorflow packages anyways
Since Tensorflow 2.x tensorflow-gpu is usually already contained in the normal tensorflow package. Source: https://www.tensorflow.org/install/gpu

The Tensorflow maintainers fucked up once so one of the versions (2.2? I don't remember) does not fully adhere to this, but the newest versions should all be fine.

I would suggest to drop tensorflow-gpu, because it won't really be maintained in the future and cause mores issues than benefit.

Is there a special reason for the tensorflow-gpu dependency @davidsebfischer ?

### PR DESCRIPTION
**Pull request by Zethson**
Added a subset of the [cookietemple cli-python template](https://github.com/cookiejar/cookietemple).

1. CI to build the package, documentation and to publish the package etc
2. fix #71 
3. Enabled dependabot
4. Added a Makefile to the documentation -> it now correctly builds all modules
5. Added issue/PR templates
6. Added Tensorflow as a hard dependency to temporarily solve #70. Let's keep #70 open until this is solved, but else many workflows would fail.
7. Added docs/requirements.txt, since this is the default location where RTD expects requirements
8. Added setup.py classifiers
9. Many more minor changes that I already forgot.

Signed-off-by: Zethson <lukas.heumos@posteo.net>

### CODE REVIEW

**Comment by Zethson** [comment]
```
Run python -c "import sfaira"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/runner/work/sfaira/sfaira/sfaira/__init__.py", line 24, in <module>
    import sfaira.models
  File "/Users/runner/work/sfaira/sfaira/sfaira/models/__init__.py", line 2, in <module>
    from .pp_layer import PreprocInput
  File "/Users/runner/work/sfaira/sfaira/sfaira/models/pp_layer.py", line 1, in <module>
    import tensorflow as tf
ModuleNotFoundError: No module named 'tensorflow'
Error: Process completed with exit code 1.
```

Package building workflows dude to #70

**Comment by Zethson** [comment]
@davidsebfischer After merging this PR you can set your PYPI_USERNAME as a repository secret and your PYPI_PASSWORD (with exactly these names!). Whenever you create a Github release then a PyPI release will automatically be done and uploaded!

File: sfaira/estimators/__init__.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from sfaira.estimators.keras import EstimatorKeras, EstimatorKerasEmbedding, EstimatorKerasCelltype
+
 try:
-    from sfaira_extension.estimators import *
+    from sfaira_extension.estimators import *  # noqa: F403
```
Comment (davidsebfischer) [inline sfaira/estimators/__init__.py:4]: what is this?

File: sfaira/models/celltype/marker.py [hunk 2]
```diff
@@ -130,7 +131,7 @@ def __init__(
         self.model_class = topology_container.model_class
         self.model_type = topology_container.model_type
         self.hyperparam = dict(
-            list(hyperpar.items()) +
+            list(hyperpar.items()) +  # noqa: W504
```
Comment (davidsebfischer) [inline sfaira/models/celltype/marker.py:134]: and here

File: sfaira/models/celltype/mlp.py [hunk 3]
```diff
@@ -106,7 +106,7 @@ def __init__(
         self.model_class = topology_container.model_class
         self.model_type = topology_container.model_type
         self.hyperparam = dict(
-            list(hyperpar.items()) +
+            list(hyperpar.items()) +  # noqa: W504
```
Comment (davidsebfischer) [inline sfaira/models/celltype/mlp.py:109]: and here

File: sfaira/models/embedding/ae.py [hunk 4]
```diff
@@ -224,7 +224,7 @@ def __init__(
         self.model_class = topology_container.model_class
         self.model_type = topology_container.model_type
         self.hyperparam = dict(
-            list(hyperpar.items()) +
+            list(hyperpar.items()) +  # noqa: W504
```
Comment (davidsebfischer) [inline sfaira/models/embedding/ae.py:227]: and here

File: sfaira/models/embedding/linear.py [hunk 5]
```diff
@@ -119,7 +119,7 @@ def __init__(
         self.model_class = topology_container.model_class
         self.model_type = topology_container.model_type
         self.hyperparam = dict(
-            list(hyperpar.items()) +
+            list(hyperpar.items()) +  # noqa: W504
```
Comment (davidsebfischer) [inline sfaira/models/embedding/linear.py:122]: and here

File: sfaira/models/embedding/vae.py [hunk 6]
```diff
@@ -247,7 +246,7 @@ def __init__(
         self.model_class = topology_container.model_class
         self.model_type = topology_container.model_type
         self.hyperparam = dict(
-            list(hyperpar.items()) +
+            list(hyperpar.items()) +  # noqa: W504
```
Comment (davidsebfischer) [inline sfaira/models/embedding/vae.py:249]: and here

File: sfaira/models/embedding/vaeiaf.py [hunk 7]
```diff
@@ -365,7 +364,7 @@ def __init__(
         self.model_class = topology_container.model_class
         self.model_type = topology_container.model_type
         self.hyperparam = dict(
-            list(hyperpar.items()) +
+            list(hyperpar.items()) +  # noqa: W504
```
Comment (davidsebfischer) [inline sfaira/models/embedding/vaeiaf.py:367]: and here

File: sfaira/models/embedding/vaevamp.py [hunk 8]
```diff
@@ -309,7 +309,7 @@ def __init__(
         self.model_class = topology_container.model_class
         self.model_type = topology_container.model_type
         self.hyperparam = dict(
-            list(hyperpar.items()) +
+            list(hyperpar.items()) +  # noqa: W504
```
Comment (davidsebfischer) [inline sfaira/models/embedding/vaevamp.py:312]: and here

File: sfaira/versions/topology_versions/external.py [hunk 9]
```diff
@@ -1 +1 @@
-from sfaira.versions.genome_versions import SuperGenomeContainer
\ No newline at end of file
+from sfaira.versions.genome_versions import SuperGenomeContainer  # noqa: W292
```
Comment (davidsebfischer) [inline sfaira/versions/topology_versions/external.py:1]: and here, missing new line?

File: versioneer.py [hunk 10]
```diff
@@ -561,15 +561,15 @@ def git_get_keywords(versionfile_abs):
         f = open(versionfile_abs, "r")
         for line in f.readlines():
             if line.strip().startswith("git_refnames ="):
-                mo = re.search(r'=\s*"(.*)"', line)
+                mo = re.search(r'=\s*"(.*)"', line)  # noqa: W605
```
Comment (davidsebfischer) [inline versioneer.py:564]: and here

File: sfaira/train/summaries.py [hunk 11]
```diff
@@ -666,26 +666,26 @@ def create_summary_tab(self):
...
                     else id_i.split("_")[3]
                     for id_i in self.run_ids
                 ],
                 "model_gs_id": ["_".join(id_i.split("_")[:(self.model_id_len + 6)]) for id_i in self.run_ids],
                 "run": self.run_ids
-            }.items()) +
-            list(dict([("train_" + m, [self.evals[x]["train"][m] for x in self.run_ids]) for m in metrics]).items()) +
-            list(dict([("test_" + m, [self.evals[x]["test"][m] for x in self.run_ids]) for m in metrics]).items()) +
-            list(dict([("val_" + m, [self.evals[x]["val"][m] for x in self.run_ids]) for m in metrics]).items()) +
+            }.items()) +  # noqa: W504
```
Comment (davidsebfischer) [inline sfaira/train/summaries.py:685]: and here

**Comment by Zethson** [comment]
The noqa statements are used to make flake8 happy:

https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html

Sometimes you want to keep a certain style, which violates flake8 -> ignore it

**Review by davidsebfischer — APPROVED** [review]

**Review by le-ander — APPROVED** [review]
Super cool stuff. Thanks a lot, Lukas!

**Pull request merged**